### PR TITLE
chore: Add Swap Settings v1 api support

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/api/buildSwapTransaction.ts [hunk 1]
```diff
@@ -45,6 +45,17 @@ export async function buildSwapTransaction(
     };
   }
```
Comment (Zizzamia) [inline src/api/buildSwapTransaction.ts:47]: Let's remove those extra spaces.
Let's add tests.

File: src/api/buildSwapTransaction.ts [hunk 2]
```diff
@@ -45,6 +45,17 @@ export async function buildSwapTransaction(
...
 
+  // Adjust slippage for V1 API (aggregator)
+  // V1 expects slippage in tenths of a percent (e.g., 30 = 3%)
+  if (params.useAggregator && params.maxSlippage) {
+    apiParams = {
+      slippagePercentage: (
+        Number.parseFloat(params.maxSlippage) * 10
+      ).toString(),
+      ...apiParams,
```
Comment (Zizzamia) [inline src/api/buildSwapTransaction.ts:55]: Why are we doing this? this seems odd.

You just can do
`apiParams.slippagePercentage = potato`
no need all the extra code

Comment (cpcramer) [reply src/api/buildSwapTransaction.ts:55]: Updated to:
```

    let slippage = Number(params.maxSlippage);
  
    slippage = slippage * 10;
    
    apiParams.slippagePercentage = slippage.toString();
```

File: src/api/buildSwapTransaction.test.ts [hunk 3]
```diff
@@ -283,7 +283,7 @@ describe('buildSwapTransaction', () => {
     expect(sendRequest).toHaveBeenCalledTimes(1);
     expect(sendRequest).toHaveBeenCalledWith(CDP_GET_SWAP_TRADE, [
       {
-        slippagePercentage: '3',
```
Comment (cpcramer) [inline src/api/buildSwapTransaction.test.ts:286]: @0xAlec Has this test been broken? It seems like we've been calling the v1 api with the wrong slippagePercentage (not multiplying the value by 10?)

File: src/api/buildSwapTransaction.test.ts [hunk 4]
```diff
@@ -201,7 +201,7 @@ describe('buildSwapTransaction', () => {
 
   it('should return a swap with an approve transaction', async () => {
     const mockParams = {
-      useAggregator: true,
```
Comment (cpcramer) [inline src/api/buildSwapTransaction.test.ts:204]: This is a broken test, useAggregator should be set to false.

File: src/api/buildSwapTransaction.test.ts (hunk 3 above)
Comment (cpcramer) [reply src/api/buildSwapTransaction.test.ts:286]: confirmed that this test has been broken

File: src/api/buildSwapTransaction.ts [hunk 5]
```diff
@@ -30,19 +30,19 @@ export async function buildSwapTransaction(
...
   }
-  let apiParams = apiParamsOrError as SwapAPIParams;
+  const apiParams = apiParamsOrError as SwapAPIParams;
 
   if (!params.useAggregator) {
-    apiParams = {
-      v2Enabled: true,
-      ...apiParams,
-    };
+    apiParams.v2Enabled = true;
```
Comment (Zizzamia) [inline src/api/buildSwapTransaction.ts:36]: yeah sorry, my bad, let's go back to the 
```
{
      v2Enabled: true,
      ...apiParams,
    };
```
approach

Comment (cpcramer) [reply src/api/buildSwapTransaction.ts]: done

Comment (cpcramer) [reply src/api/buildSwapTransaction.ts]:
```
  let apiParams = apiParamsOrError as SwapAPIParams;
  if (!params.useAggregator) {
    apiParams = {
      v2Enabled: true,
      ...apiParams,
    };
  }
  if (params.maxSlippage) {
    let slippagePercentage = params.maxSlippage;
    // Adjust slippage for V1 API (aggregator)
    // V1 expects slippage in tenths of a percent (e.g., 30 = 3%)
    if (params.useAggregator) {
      slippagePercentage = (Number(params.maxSlippage) * 10).toString();
    }
    apiParams = {
      slippagePercentage,
      ...apiParams,
    };
  }
```

File: src/api/buildSwapTransaction.ts [hunk 6]
```diff
@@ -39,8 +39,14 @@ export async function buildSwapTransaction(
     };
   }
   if (params.maxSlippage) {
+    let slippagePercentage = params.maxSlippage;
+    // Adjust slippage for V1 API (aggregator)
+    // V1 expects slippage in tenths of a percent (e.g., 30 = 3%)
+    if (params.useAggregator) {
```
Comment (Zizzamia) [inline src/api/buildSwapTransaction.ts:45]: I still don't follow, is `useAggregator: true` what everyone uses at the moment?

If that so, we should not change this code.

Comment (cpcramer) [reply src/api/buildSwapTransaction.ts:45]: There are two different APIs, V1 [wallet/swap-service](https://github.cbhq.net/cloud/swap-router) and V2 [cloud/swap-router](https://github.cbhq.net/cloud/swap-router). 

V1:
- `useAggregator: true`
- Expects slippage in tenths of a percent (e.g., 30 = 3%) which is why we must multiply maxSlippage by 10 before sending the request. 

V2:
- `useAggregator: false`
- Expects slippage in a percent, which is why we do not have to adjust the maxSlippage value.

PartyDOA is the only known customer who is using the maxSlippage parameter.

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**